receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go

```diff
@@ -75,11 +97,19 @@ func (s *scraper) ScrapeMetrics(_ context.Context) (pdata.MetricSlice, error) {
 		usages = append(usages, &deviceUsage{partition.Device, usage})
 	}
 
-	if len(usages) > 0 {
+	// filter devices by name
+	filteredUsages := make([]*deviceUsage, 0, len(usages))
+	for _, usage := range usages {
+		if s.includeDevice(usage.deviceName) {
+			filteredUsages = append(filteredUsages, usage)
+		}
+	}
```

Comment:
this can be optimized:

```golang
filteredUsages := usage
s.includeFS != nil || s.excludeFS != nil {
	filteredUsages := make([]*deviceUsage, 0, len(usages))
	for _, usage := range usages {
		if s.includeDevice(usage.deviceName) {
			filteredUsages = append(filteredUsages, usage)
		}
	}
}
```